File: core/src/main/java/com/serious/service/exception/ServiceRegistryException.java
package com.serious.service.exception;
/*
 * @COPYRIGHT (C) 2016 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * @author Andreas Ernst
 */
public class ServiceRegistryException extends RuntimeException {
    // ServiceException

    public ServiceRegistryException(String message, Object... args) {
        super(String.format(message, args));
    }

    ServiceRegistryException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: core/src/main/java/com/serious/service/exception/ServiceRegistryException.kt
package com.serious.service.exception
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

/**
 * Any exception that occurs during setup of the service framework.
 */
class ServiceRegistryException : RuntimeException {
    constructor(message: String, vararg args: Any?) : super(String.format(message, *args))

    internal constructor(message: String, cause: Throwable?) : super(message, cause)
}